editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/adopters/configurators/MavenTemplateConfigurator.java

```diff
@@ -44,13 +45,27 @@ public boolean configure(IProject project, CommonNewProjectMetaData metadata, IP
 		if (ok) {
 			// by default add the maven nature
 			ok = configureMavenNature(project, subMonitor.newChild(1));
+			new BuildAndRefreshJobWaiterUtil().waitJob(subMonitor.newChild(1));
 		}
-		
-		if (ok && !Strings.isBlank(metadata.getCamelVersion())) {
-			// by default configure the version of camel used in the pom.xml
-			ok = MavenUtils.configurePomCamelVersion(project, metadata, metadata.getCamelVersion(), subMonitor.newChild(1));
+
+		// to not trigger too many events while configuring the pom  we reuse the model
+		Model m2model = new CamelMavenUtils().getMavenModel(project);
+
+		try {
+			if (ok) {
+				// by default add staging repos if option enabled
+				MavenUtils.manageStagingRepositories(m2model);
+			}
+			subMonitor.setWorkRemaining(2);
+			
+			if (ok && !Strings.isBlank(metadata.getCamelVersion())) {
+				// by default configure the version of camel used in the pom.xml
+				MavenUtils.configurePomCamelVersion(project, m2model, metadata, metadata.getCamelVersion(), subMonitor.newChild(1));
```

Comment:
why removing the "ok" check on this method? For this method it is returning if the method was in success or not

Reply:
yeah, and the result was never evaluated nor used.

Reply:
if not used:
- and not planned to be used: we should remove the return value
- and potentially planned: we need to continue to return the value so that there is not a bug when it will be used
So in this PR I would prefer to keep the return value and if  you think that it will never be used, create a PR to don't return this boolean value

Reply:
It is currently not used. That doesn't mean it will stay like this forever. If we decide to use the return value then its easy to reintroduce the ok=. I am not going to change the signature of the called method. That does not make any sense.

Reply:
>If we decide to use the return value then its easy to reintroduce the ok=

This is exactly what is hard. How can the next developer that might not be us guess that he/she needs to readd the "ok="? And even if it is me in it is one month I won't be able to remind it.